pkg/config/dynamic/tcp_config.go

```diff
@@ -72,8 +72,9 @@ type TCPServersLoadBalancer struct {
 	// connection, to close the reading capability as well, hence fully terminating the
 	// connection. It is a duration in milliseconds, defaulting to 100. A negative value
 	// means an infinite deadline (i.e. the reading capability is never closed).
-	TerminationDelay *int        `json:"terminationDelay,omitempty" toml:"terminationDelay,omitempty" yaml:"terminationDelay,omitempty"`
-	Servers          []TCPServer `json:"servers,omitempty" toml:"servers,omitempty" yaml:"servers,omitempty" label-slice-as-struct:"server"`
+	TerminationDelay     *int        `json:"terminationDelay,omitempty" toml:"terminationDelay,omitempty" yaml:"terminationDelay,omitempty"`
+	ProxyProtocolVersion string      `json:"proxyProtocolVersion,omitempty" toml:"proxyProtocolVersion,omitempty" yaml:"proxyProtocolVersion,omitempty"`
```

Comment:
After discussing with the rest of the team, we think the attribute `ProxyProtocolVersion` might be misleading. Since it contains "version" in its name people might understand it has: 
> Proxy protocol is always enabled, `ProxyProtocolVersion` is just here for overriding the default version. 

To disambiguate the naming we suggest to change this field into a struct like this:

```suggestion
	ProxyProtocol *ProxyProtocol      `json:"proxyProtocol,omitempty" toml:"proxyProtocol,omitempty" yaml:"proxyProtocol,omitempty"`
```

And the `ProxyProtocol` struct would be defined as:

```go
type ProxyProtocol struct {
  Version string
}
```

Doing this would lead people to think, proxy protocol is disabled by default. This way we can even define a `SetDefaults` for the `ProxyProtocol` type to set its version to "2" by default. Then if the protocol evolves in the future and allow for more configuration than just the version, we will be already prepared.

Reply:
Just a question:

```
// SetDefaults Default values for a TCPServersLoadBalancer.
func (l *TCPServersLoadBalancer) SetDefaults() {
	defaultTerminationDelay := 100 // in milliseconds
	l.TerminationDelay = &defaultTerminationDelay
	l.ProxyProtocol = &ProxyProtocol{Version: "2"}
}
```

this would enable proxyprotocol v2 by default? It should be disabled by default

Reply:
```go
func (p *ProxyProtocol) SetDefaults() {}
```

Reply:
Where should ProxyProtocol.SetDefaults() be executed? I defined the function but its not called yet.

Reply:
`SetDefaults` is executed automatically when the provider constructs the dynamic configuration. This is done when providers call `parser.Fill` from the `traefik/paerser` package. You can find more information on 
https://github.com/traefik/paerser/blob/d69e93b8aaa12317e7e8f609b490232d8ad5f175/parser/element_fill.go#L104